plugins/sam3dj.js

```diff
@@ -1,353 +1,401 @@
-                    warning.form.border = {
-                        type: 'info',
-                        text: '... are actually a Mesh and not a Cube',
-                    }
-    
-                warning.form.nextstep = {
-                    type: 'info',
-                    text: '*Press \'Confirm\' if you still want to proceed.*',
-                }
-    
-                warning.show();
-            // Or just export if everything is right :)
-            } else
-                popConfigModel.show();
-        }
-    });
-    
-    Plugin.register('sam3dj', pluginProperties);
-    })();
+/** sam3DJ | aka. Sammy's .3DJ (format implementation)
+ *  ==================================================
+ *  Author:     Sammy L. Koch (1Turtle)
+ *  Licence:    Apache-2.0 (see https://www.apache.org/licenses/LICENSE-2.0.txt)
+*/
+(function () {
+	const pluginProperties = {
+		title: 'sam3DJ',
+		author: '1Turtle',
+		tags: ['Minecraft: Java Edition', 'Format', 'Exporter'],
+		version: '1.0.2',
+		variant: 'both',
+		description: 'Generate 3D prints in MC (Java Edition) via the .3DJ format within the SC-Peripherals mod!',
+		about: "This plugin allows you to **export a model into the .3DJ format**.  \nThe .3DJ format is _mostly_ used with the `3D Printer` from the **SC-Peripherals** Java Edition mod.  \n  \n**Tip:** Cubes in a `state_on` folder will be _only_ displayed whenever the model is toggled on!    \n> **Note:** _The format does not support the following attributes:_  \n> * Rotated cubes  \n> * One cube with multiple textures  \n> * Cubes outside the given block space *(16x16x16 Grid / 1x1x1 Block)*  \n  \nTry to avoid those things, or else you might run into weird results.  \nMeshes are also not supported due to the limitations of the format and Minecraft.\n> For the full format documentations, see the [SwitchCraft3 Wiki](https://docs.sc3.io/whats-new/sc-peripherals.html#_3dj-format)  \n",
+		icon: 'print',
+		onload() {
+			MenuBar.addAction(btnExport, 'file.export');
+		},
+		onunload() {
+			MenuBar.removeAction('file.export.export_3dj');
+		}
+	};
+	const SC_TEXTURE_BLANK = 'sc-peripherals:block/white';
+
+	/**
+	 * Returns a fitting texture for a given cube.
+	 * @param {*} cube The cube, which holds the wanted texture.
+	 * @returns The wanted texture, fitting to the cube. (+ some booleans) in a object.
+	 */
+	function getSingleTexture(cube, scan) {
+		let faces = 0;
+		let missing = false;
+		let texture = SC_TEXTURE_BLANK;
+
+		for (let dir in cube.faces) { //in ['north', 'east', 'south', 'west', 'up', 'down']
+			let tmp = cube.faces[dir].getTexture();
+			if (tmp === undefined) {
+				missing = true;
+				continue;
+			}
+			let logOutput = ""
+			// texture: <namespace>:<path>
+			let namespace = ""
+			let id = ""
+			if (tmp.name.includes(":")) {
+				id = tmp.name
+			} else if (tmp.name.includes("/")) {
+				id = "minecraft:" + tmp.name
+			}
+			if (id === "") {
+				if (cube.name.includes(":")) {
```

Comment:
This whole section seems questionable to me, if this condition is met, the other renaming logic further down does never run.
I assume file extension removal should always be performed, and be further down, with the other renaming logic.
Though I am not farmiliar with the 3DJ format, therefore please correct me if I am incorrect in my assumptions.